scripts/reserved-names/deny-labels.ts

```
@@ -59,37 +69,72 @@ export const addReservedLabels = (
     });
 };
 
-// Check if any label files exist
-export const hasLabelFilesToProcess = (): boolean => {
-    const files = ['./init/labels-to-block.txt', './init/labels-to-reserve.txt'];
+// Check if a label file exists and has at least one valid label
+export const hasLabelsInFile = (filePath: string): boolean => {
+    if (!fs.existsSync(filePath)) return false;
+    try {
+        const labels = parseLabelsFile(filePath);
+        return labels.length > 0;
+    } catch {
+        return false;
+    }
+};
 
-    return files.some((path) => fs.existsSync(path));
+// Type for package info
+export type PackageInfoForLabels = {
+    packageId: string;
+    iotaNamesObjectId: string;
+    adminCap: string;
 };
 
-// Reserve and block labels from files if provided
-export const processLabelFiles = (
+// Process a batch of labels for blocking or reserving
+export function processLabelsBatch(
     tx: Transaction,
-    packageId: string,
-    iotaNamesObjectId: string,
-    adminCap: string,
-): boolean => {
-    const files = [
-        { path: './init/labels-to-block.txt', action: 'Blocking', fn: addBlockedLabels },
-        { path: './init/labels-to-reserve.txt', action: 'Reserving', fn: addReservedLabels },
-    ];
-
-    let hasLabels = false;
+    packageInfo: PackageInfoForLabels,
+    action: 'block' | 'reserve',
+    labels: string[],
+): void {
+    const { packageId, iotaNamesObjectId, adminCap } = packageInfo;
+    if (action === 'block') {
+        addBlockedLabels(tx, packageId, iotaNamesObjectId, adminCap, labels);
+    } else {
+        addReservedLabels(tx, packageId, iotaNamesObjectId, adminCap, labels);
+    }
```

maybe better to throw an error if the action is neither `block` nor `reserve`, or is TS already catching that?

---

typescript validates this at build time I believe